patches/express-graphql+0.9.0.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/node_modules/express-graphql/index.js b/node_modules/express-graphql/index.js
+index 609aba7..572a7d1 100644
+--- a/node_modules/express-graphql/index.js
++++ b/node_modules/express-graphql/index.js
+@@ -167,6 +167,12 @@ function graphqlHTTP(options) {
+         };
+       }
+     }).then(result => {
++      const { errors } = result
++      if (errors) {
++        if (errors.find(error => error.message === "Braze rate limit reached")) {
++          response.statusCode = 429;
```

Comment:
This implementation is all very low code, but is there no way to do this in a more vanilla way? 

Another thought is that, calls made from braze use a custom directive, and from there [we can inspect](https://github.com/artsy/metaphysics/blob/main/src/directives/principleField/principalFieldDirectiveExtension.ts#L15-L17) various bits and redirect behavior. 

(It seems like the the foundation of the implementation might be off if we need to patch and assign status codes from within the patch.)

Reply:
So yea - I don't have much insight in what can be done in Braze - @cavvia has sort-of looked at that I think. I believe Braze requires a 429 status code for 'connected content' in order to know to fall-back and retry.

If there's a way to customize that on the Braze side (such as be able to inspect or introspect something about the returned data) - all the better! But that research needs to happen on the Braze side, otherwise we would need to return a 429.

And the reason this implementation is kind-of screwy is b/c this isn't very GraphQL - where we almost always return a 200. That's why I needed to reach into the lib to pass that on. I totally agree with you it would be nice to solve this in a different, even more GraphQL-appropriate way - but that just brings me back to the point that we need to figure out if that's _possible_ from the Braze side. From what I've been told (I haven't investigated on the Braze side much), we need a 429 response code :{

Reply:
(It's also possible that we can implement this not-very-GraphQL-like 429 thing in a different way that _doesn't_ require patching the lib. This was just my initial draft/attempt as part of #ff. I could give this another shot to see if there are other ways - wrapping the entire `express-graphql` MP app in a middleware that, based on a custom extension, if there's a Braze 429 present can just return an empty body w/ appropriate status code.

That's a bit verbose/roundabout, but could be a setup that works and avoids patching.

This was just my initial stab at getting something that might work for the Braze/DOS issue.

Reply:
I think ^ would be totally fine! Middleware intersection like this seems appropriate and correct.

Reply:
I don't believe there's any retry logic customisation we can do on the Braze side, they [support retries](https://www.braze.com/docs/user_guide/personalization_and_dynamic_content/connected_content/connected_content_retries) with exponential backoff and you can just adjust a rate limit per campaign there. I think it's safe to assume they'll behave correctly when receiving a 429 as they must get rate limited all the time.